clients/go/cmd/zbctl/internal/commands/status.go

```diff
@@ -117,3 +128,12 @@ func healthToString(health pb.Partition_PartitionBrokerHealth) string {
 		return unknownState
 	}
 }
+
+func printTopologyJSON(resp *pb.TopologyResponse) error {
```

Comment:
I have to move it to a separate function because JSONPB marshals `int64` as Go `string`. See: https://developers.google.com/protocol-buffers/docs/proto3#json

Reply:
Do I understand correctly this means we cannot just replace the implementation of `printJSON()` in `commands/root.go`?

Reply:
Correct. For example, what the test show:
```
        --- FAIL: TestZbctlWithInsecureGateway/TestCommonCommands/deploy_workflow (0.97s)
            main_test.go:150: deploy workflow: diff (-want +got):
                  []string{
                        "{",
                -       `  "key": 2251799813685250,`,
                +       `  "key": "2251799813685251",`,
                        `  "workflows": [`,
                        "    {",
                        `      "bpmnProcessId": "process",`,
                        `      "version": 1,`,
                -       `      "workflowKey": 2251799813685345,`,
                +       `      "workflowKey": "2251799813685249",`,
                        `      "resourceName": "model.bpmn"`,
                        "    },",
                        "    {",
                        `      "bpmnProcessId": "jobProcess",`,
                        `      "version": 1,`,
                -       `      "workflowKey": 2251799813685346,`,
                +       `      "workflowKey": "2251799813685250",`,
                        `      "resourceName": "job.bpmn"`,
                        "    }",
                        ... // 3 identical elements
                  }

```
After this diff:
```diff
Index: clients/go/cmd/zbctl/internal/commands/root.go
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- clients/go/cmd/zbctl/internal/commands/root.go	(revision 0d4eff5158bd87c7c7ae224a7bee9ed3953c7e7a)
+++ clients/go/cmd/zbctl/internal/commands/root.go	(date 1606800703955)
@@ -15,9 +15,10 @@
 package commands
 
 import (
-	"encoding/json"
 	"fmt"
 	"github.com/zeebe-io/zeebe/clients/go/pkg/zbc"
+	"google.golang.org/protobuf/encoding/protojson"
+	"google.golang.org/protobuf/reflect/protoreflect"
 	"os"
 	"strconv"
 	"strings"
@@ -201,8 +202,9 @@
 	}
 }
 
-func printJSON(value interface{}) error {
-	valueJSON, err := json.MarshalIndent(value, "", "  ")
+func printJSON(value protoreflect.ProtoMessage) error {
+	m := protojson.MarshalOptions{EmitUnpopulated: true, Indent: "  "}
+	valueJSON, err := m.Marshal(value)
 	if err == nil {
 		fmt.Println(string(valueJSON))
 	}

```

Reply:
Thanks for this, very clear.

@MiguelPires This change is backwards compatible this way, but it also means that json marshalling is done differently for this response compared to others. Should we keep backwards compatibility and introduce this inconsistency or should we break backwards compatibility and go for consistency? I'd love to hear what you think

Reply:
I think right now our priority is maintaining backwards compatibility but @npepinpe can answer that better

Reply:
Backwards compat is pretty important, but primarily in the broker - in the client it really depends on what the impact is. I'm not really into the current topic, so how is backwards compatibility broken here? (let's ignore that I'm confused as to why int64 is converted to string :sweat_smile:)

Reply:
```
                -       `  "key": 2251799813685250,`,
                +       `  "key": "2251799813685251",`,
```
JSONPB serializes the int64 fields as string and the standard one doesn't

Reply:
Dumb question, but why is it using JSONPB and not just the standard one? Anyway, I assume we have to? And there's no way to change this behaviour? Because that's pretty weird.

At any rate, I imagine we're not going to use JSONPB for the other commands, right? So this only affects the topology? And only if we specify `--json`? Where does the backwards compatibility breaks?

Reply:
Yes, the standard json printer doesn't do well with protobuf because of the enums. The issue is not with jsonpb, specifying int64 integers as string is part of the protobuf spec. From what I read, it's because in some browsers javascript had trouble reading int64s (that's something I read online, I didn't do any research on it).
Actually, if we can break compatibility I would suggest adding this flag everywhere and either print things in a human-readable or using JSONPB. It would avoid issues like https://github.com/zeebe-io/zeebe/issues/5888

Reply:
> Actually, if we can break compatibility I would suggest adding this flag everywhere and either print things in a human-readable or using JSONPB

Let's do that, but I think this is maybe outside of scope here. Let's create a follow up issue, and for now let's implement this flag only for this command. wdyt?